File: API/WorkoutTracker.API/Models/ExerciseInWorkoutDTO.cs
using System;
using WorkoutTracker.Domain.Exercises;

namespace WorkoutTracker.API.Models
{
    public record ExerciseInWorkoutDTO(
        int Id, Guid ExerciseId, string ExerciseName, short NumberOfSets, SetType SetType, ResistanceType ResistanceType);
}

File: workout-tracker/src/app/workouts/_models/exercise-in-workout-dto.ts
import { ResistanceType } from "../workout/_enums/resistance-type";
import { SetType } from "../workout/_enums/set-type";

export class ExerciseInWorkoutDTO {
  id: number = 0;
  exerciseId: string = '';
  exerciseName: string = ''; //TODO: Revisit. Is this being used?
  numberOfSets: number = 0;
  setType: SetType = SetType.Repetition;
  resistanceType: ResistanceType = ResistanceType.BodyWeight;
}